src/pal/tools/gen-buildsys-gcc.sh

```diff
@@ -80,55 +82,31 @@ for i in "$@"; do
     esac
 done
 
-OS=`uname`
+OS=$(uname)
 
 locate_gcc_exec() {
   if command -v "$gcc_prefix$1$desired_gcc_version" > /dev/null 2>&1
   then
-    echo "$(command -v $gcc_prefix$1$desired_gcc_version)"
+    command -v "$gcc_prefix$1$desired_gcc_version"
   elif command -v "$gcc_prefix$1" > /dev/null 2>&1
   then
-    echo "$(command -v $gcc_prefix$1)"
+    command -v "$gcc_prefix$1"
   else
     exit 1
   fi
 }
 
-gcc_ar="$(locate_gcc_exec ar)"
-[ $? -eq 0 ] || { echo "Unable to locate gcc-ar"; exit 1; }
+if ! gcc_ar="$(locate_gcc_exec ar)"; then { echo "Unable to locate gcc-ar"; exit 1; } fi
 
-if [ -z "$CC" ]; then
```

Comment:
These changes also break my current cross compilation set up. The assumption below is that I should be able to set CC, CXX etc. environment variables to my compiler with the compiler setting Iike. 

Can you maybe change locate_gcc_exec to satisfy this and still keep it clean like you are doing below?

Reply:
IMO, we should use a custom variable or command line option, instead of relying on `CC`; which is commonly set in many user environments.

`CC` gets internally set in clang and gcc buildsys scripts for CMake and does not affect the user environment.

Something like: `CLR_C_COMPILER` and `CLR_CXX_COMPILER`. Will that work?

Reply:
I just tested cross compilation. If you fix the issue below, there is no other issue.

Reply:
Taking it back. It still thinks that it is compiling x64_64. I'd like to keep the original environment variables if possible.

Reply:
Error while cross-compiling arm64.

src/pal/src/libunwind/src/x86_64/Gos-linux.c:123:50: error: ‘mcontext_t {aka struct <anonymous>}’ has no member named ‘gregs’; did you mean ‘regs’?
     case UNW_X86_64_RDX: addr = &uc->uc_mcontext.gregs[REG_RDX]; break;

Reply:
OK, I have pushed a commit. Automatic variables are:

* `DOTNET_CC`
* `DOTNET_CXX`
* `DOTNET_NM`
* `DOTNET_LINK`
* `DOTNET_AR`
* `DOTNET_OBJDUMP`
* `DOTNET_OBJCOPY`

Can you please try with:

```sh
DOTNET_CC=$(command -v gcc)              \
DOTNET_CXX=$(command -v g++)             \
DOTNET_NM=$(command -v YourNM)           \
DOTNET_AR=$(command -v YourAR)           \
DOTNET_LINK=$(command -v YourLINK)       \
DOTNET_OBJDUMP=$(command -v YourOBJDUMP) \
DOTNET_OBJCOPY=$(command -v YourOBJCOPY) \
./build.sh x64 checked gcc
```

to see if it picks up your custom tool chain.